tests-js/protocol/WithdrawFeesAndInterest.test.js

```diff
@@ -0,0 +1,178 @@
+const { expect } = require("chai");
+const { BN } = require("@openzeppelin/test-helpers");
+const { increaseTime, blockNumber } = require("../Utils/Ethereum.js");
+
+const {
+	getSUSD,
+	getRBTC,
+	getWRBTC,
+	getBZRX,
+	getLoanTokenLogic,
+	getLoanToken,
+	getLoanTokenLogicWrbtc,
+	getLoanTokenWRBTC,
+	loan_pool_setup,
+	set_demand_curve,
+	getPriceFeeds,
+	getSovryn,
+	lend_to_pool,
+	open_margin_trade_position,
+} = require("../Utils/initializer.js");
+
+contract("ProtocolWithdrawFeeAndInterest", (accounts) => {
+	let owner;
+	let sovryn, SUSD, WRBTC, RBTC, BZRX, loanToken, loanTokenWRBTC, priceFeeds;
+
+	before(async () => {
+		[owner] = accounts;
+	});
+
+	beforeEach(async () => {
+		SUSD = await getSUSD();
+		RBTC = await getRBTC();
+		WRBTC = await getWRBTC();
+		BZRX = await getBZRX();
+		priceFeeds = await getPriceFeeds(WRBTC, SUSD, RBTC, sovryn, BZRX);
+
+		sovryn = await getSovryn(WRBTC, SUSD, RBTC, priceFeeds);
+
+		const loanTokenLogicStandard = await getLoanTokenLogic();
+		const loanTokenLogicWrbtc = await getLoanTokenLogicWrbtc();
+		loanToken = await getLoanToken(loanTokenLogicStandard, owner, sovryn, WRBTC, SUSD);
+		loanTokenWRBTC = await getLoanTokenWRBTC(loanTokenLogicWrbtc, owner, sovryn, WRBTC, SUSD);
+		await loan_pool_setup(sovryn, owner, RBTC, WRBTC, SUSD, loanToken, loanTokenWRBTC);
+	});
+
+	describe("Tests withdraw fees andn interest ", () => {
+		it("Test withdraw accrued interest", async () => {
+			// prepare the test
+			await set_demand_curve(loanToken);
+			await lend_to_pool(loanToken, SUSD, owner);
+			const [loan_id] = await open_margin_trade_position(loanToken, RBTC, WRBTC, SUSD, owner);
+
+			let num = await blockNumber();
+			let lastBlock = await web3.eth.getBlock(num);
+			const initial_block_timestamp = lastBlock.timestamp;
+
+			const loan = await sovryn.getLoan(loan_id);
+			const lender = loanToken.address;
+
+			// Time travel
+			const time_until_loan_end = loan["endTimestamp"] - initial_block_timestamp;
+			await increaseTime(time_until_loan_end);
+
+			num = await blockNumber();
+			lastBlock = await web3.eth.getBlock(num);
+			const second_block_timestamp = lastBlock.timestamp;
```

Comment:
orig: second_block_timestamp = chain[-1].timestamp
don't we need to take the previous block ts, not current?

Reply:
No it means current block, tested with hardhat and brownie tests